scripts/eft-story-wiki.py

```
@@ -0,0 +1,88 @@
+    result = {}
+    for cid, title in PAGES.items():
+        try:
+            wt = fetch_wikitext(title)
+            objs = parse_objectives(wt)
+            result[cid] = objs
+            n_opt = sum(1 for o in objs if o["optional"])
+            print(f"  {cid:22s} {len(objs):3d} objectives ({n_opt} optional)", file=sys.stderr)
+        except Exception as e:  # noqa: BLE001
+            print(f"  {cid:22s} FAILED: {e}", file=sys.stderr)
+            result[cid] = []
+    json.dump(result, open("/tmp/wiki-objectives.json", "w"), indent=2, ensure_ascii=False)
```

Ruff (0.15.20)

[warning] Use a context manager for opening files

(SIM115)